lib/travis/model/job/queue.rb

```diff
@@ -30,6 +32,25 @@ def queues
       def default
         @default ||= new(Travis.config.default_queue)
       end
+
+      def sudo_detected?(config)
+        config.values_at(*custom_stages).compact.map do |value|
+          Array(value).reject { |s| s =~ /\s*#.*/ }.map { |s| s =~ /\b(sudo|ping)\b/ }.any?
```

Comment:
We should extract these command names to be configurable. As it is, `/\b(sudo|ping)\b/` seems too magical.